### ISSUE TITLE
[Adjust Rule] SIM118: for key in list(dict.keys) can be used if the loop removes keys

### ISSUE THREAD

**Issue by MartinThoma** [MartinThoma/flake8-simplify#46]
## Desired change

* **Rule(s)**: SIM118
* **Adjustment**: Don't trigger if keys are removed within the loop

## Explanation

The change would actually introduce a bug

## Example

This is an example where the mentioned rule(s) would currently be suboptimal:

```python
# Before
for key in list(dict.keys()):
    if some_property(key):
        del dict[key]

# After (wrong?)
for key in dict:
    if some_property(key):
        del dict[key]
```

**Comment by MartinThoma** [comment]
```
        For(
            target=Name(id='key', ctx=Store()),
            iter=Call(
                func=Attribute(
                    value=Name(id='dict', ctx=Load()),
                    attr='keys',
                    ctx=Load(),
                ),
                args=[],
                keywords=[],
            ),
            body=[
                If(
                    test=Compare(
                        left=Name(id='key', ctx=Load()),
                        ops=[In()],
                        comparators=[
                            List(
                                elts=[
                                    Constant(value='a', kind=None),
                                    Constant(value='c', kind=None),
                                ],
                                ctx=Load(),
                            ),
                        ],
                    ),
                    body=[
                        Delete(
                            targets=[
                                Subscript(
                                    value=Name(id='dict', ctx=Load()),
                                    slice=Name(id='key', ctx=Load()),
                                    ctx=Del(),
                                ),
                            ],
                        ),
                    ],
                    orelse=[],
                ),
            ],
            orelse=[],
            type_comment=None,
        )
```

**Comment by MartinThoma** [comment]
Delete(
                            targets=[
                                Subscript(
                                    value=Name(id='dict', ctx=Load()),
                                    slice=Name(id='key', ctx=Load()),
                                    ctx=Del(),
                                ),
                            ],
                        ),
